### PR TITLE
feat: Create Inbox Message snippet

### PR DESCRIPTION
**Pull request by seebergs**
- [ ] I have added automatic tests where applicable.
- [ ] The PR contains a description of what has been changed.
- [ ] The description contains manual test instructions.
- [ ] The PR contains updates to the release notes.
- [ ] I have verified that the documentation is still up to date.

## Changes
Initial commit

### CODE REVIEW

**Comment by hdd** [comment]
@seebergs please add requirements.txt to the folder , likely in need of just ftrack-python-api

File: python/tools/create_inbox_message/create_inbox_message.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# :coding: utf-8
+# :copyright: Copyright (c) 2024 ftrack
+
+import ftrack_api
+
+session = ftrack_api.Session()
+
+recipients_user_names = ['<username>']
```
Comment (hdd) [inline python/tools/create_inbox_message/create_inbox_message.py]: @seebergs maybe better use a query to get any other user to pass in to send the message , so can be run and tested by the users.

File: python/tools/create_inbox_message/create_inbox_message.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# Add recipients
+for user in recipients:
+    recipient = session.create('Recipient', {
+        'note_id': note['id'],
+        'resource_id': user['id']
+    })
+    note['recipients'].append(recipient)
+
+# Persist changes to database / send "internal email"
+session.commit()
```
Comment (hdd) [inline python/tools/create_inbox_message/create_inbox_message.py]: please add a way to run as standalone (windows) using the : if __name__ == '__main__' as for other recipes:
https://github.com/ftrackhq/ftrack-recipes/blob/main/python/actions/create-report-0.0.0/hook/create_report.py#L232

File: python/tools/create_inbox_message/create_inbox_message.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# :coding: utf-8
+# :copyright: Copyright (c) 2024 ftrack
+
+import ftrack_api
+
```
Comment (hdd) [inline python/tools/create_inbox_message/create_inbox_message.py:6]: @seebergs please import and use logging to notify user about action success.

File: python/tools/create_inbox_message/create_inbox_message.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+session = ftrack_api.Session()
+
+recipients_user_names = ['<username>']
+message_content = 'Message body'
```
Comment (hdd) [inline python/tools/create_inbox_message/create_inbox_message.py]: @seebergs Change this to something like "Example Message Content from recipe" so is easy to spot.

File: python/tools/create_inbox_message/create_inbox_message.py [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+# Query author and recipients. Note that the message will not appear if the
+# author and recipient is the same user.
+author = session.query(f"User where username is {session.api_user}").one()
+recipients = session.query(
+    "User where username in ('{}')".format("','".join(recipients_user_names))
```
Comment (hdd) [inline python/tools/create_inbox_message/create_inbox_message.py]: @seebergs as we are in py3 land, better use new f string formatting as you did in the previous line

**Review by hdd — COMMENTED** [review]
few minor changes

File: python/tools/create_inbox_message/create_inbox_message.py (hunk 1 above)
Comment (seebergs) [reply python/tools/create_inbox_message/create_inbox_message.py]: Can you explain this in more detail? You mean not providing a list of usernames, rather something more generic?

File: python/tools/create_inbox_message/create_inbox_message.py (hunk 2 above)
Comment (seebergs) [reply python/tools/create_inbox_message/create_inbox_message.py]: Is this really needed for this recipe? This is a snippet that runs when you run the script so don't see the reason to use that line, and it's not something that needs a connection to the event hub or so. Is that line needed for Windows?

**Comment by seebergs** [comment]
> @seebergs please add requirements.txt to the folder , likely in need of just ftrack-python-api

Added.

File: python/tools/create_inbox_message/create_inbox_message.py (hunk 3 above)
Comment (seebergs) [reply python/tools/create_inbox_message/create_inbox_message.py:6]: Added.

File: python/tools/create_inbox_message/create_inbox_message.py (hunk 4 above)
Comment (seebergs) [reply python/tools/create_inbox_message/create_inbox_message.py]: Changed.

File: python/tools/create_inbox_message/create_inbox_message.py (hunk 5 above)
Comment (seebergs) [reply python/tools/create_inbox_message/create_inbox_message.py]: Good idea, added.

File: python/tools/create_inbox_message/create_inbox_message.py (hunk 1 above)
Comment (hdd) [reply python/tools/create_inbox_message/create_inbox_message.py]: yes, simply so users can run the code as first thing without having to change anything.
It should be matter of picking any other users which is not the one sending the message.

File: python/tools/create_inbox_message/create_inbox_message.py (hunk 2 above)
Comment (hdd) [reply python/tools/create_inbox_message/create_inbox_message.py]: Yeap, that's for windows....

**Review by hdd — CHANGES_REQUESTED** [review]

**Review by hdd — APPROVED** [review]

**Pull request merged**